poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd

```diff
@@ -0,0 +1,166 @@
+data_df <- open_dataset(data_bucket)
+```
+
+# Extracting monitoring sites and their coordinates
+
+```{r sites_bin}
+sites_shp <- data_df %>% 
+  #We select unique sites included in the dataset
+  distinct(site, subsite, lon, lat) %>%
+  #This will load them into memory
+  collect()
+
+#Creating shapefile of unique locations
+sites_shp <- sites_shp %>% 
+  #Creating column to identify deployment location
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA)) %>%
+  #Turning into shapefile
+  st_as_sf(coords = c("lon", "lat"), crs = 4326)
+
+#Checking results
+glimpse(sites_shp)
+```
+# Identifying AIMS sites within GBR feature of interest
+In this section, we will use the functions included in the [useful_spatial_functions_data_extraction](https://github.com/aodn/rimrep-examples/blob/main/poc-data-api/useful_spatial_functions_data_extraction.R) script to load GBR features of interest. We will then use the boundaries of the feature of interest to identify the names of the sites within these limits.
+
+We will use the `gbr_features` function to access the boundaries for our sites of interest. If no parameters are given to this function, it will load all features recognised by the GBRMPA. You could also provide the name of the site using the `site_name` parameter or the GBRMPA unique ID using the `site_id` parameter. You can also use both parameters at the same time in case you have the names for some sites and the IDs for others.
+
+As an example, we will use **U/N Reef** or unnamed reefs. We chose them simply because they cover a large area and will overlap with the most number of AIMS monitoring sites.
+
+```{r gbr_features}
+un_reefs <- gbr_features(site_name = "u/n reef")
+
+#Checking results
+un_reefs
+```
+## Identifying sites of interest
+Depending on the amount of polygons used to identify the sites of interest, this may take a couple of minutes.  
+  
+We will plot the original AIMS monitoring sites (black), the polygon of interest (pink) and the sites within polygons (blue) to check that sites have been correctly identified.
+
+Note that unless the AIMS site is within a polygon, it will not to be identified in our final list.
+
+```{r sites_int}
+site_list <- sites_of_interest(sites_shp, un_reefs)
+
+#Plotting results
+#AIMS monitoring sites
+sites_shp %>% 
+  ggplot()+
+  geom_sf()+
+  #Polygon of interest
+  geom_sf(inherit.aes = F, data = un_reefs, color = "pink")+
+  #AIMS polygons within area of interest
+  geom_sf(inherit.aes = F, data = site_list, color = "blue")+
+  theme_bw()
+```
+
+## Extracting temperature for monitoring site within area of interest
+
+We will now extract data only for the monitoring sites located within the GBR feature of our interest.

```

Comment:
This next step takes a lot of time to execute. Would it be better if you extract the sites of interest based on the previous step and then do the calculation locally?

Reply:
I extracted the sites and then perform the calculations and it still takes some time, but it is less than the original version.